diff --git a/absl/strings/str_cat.cc b/absl/strings/str_cat.cc
index 546b8aec..30cf50fe 100644
--- a/absl/strings/str_cat.cc
+++ b/absl/strings/str_cat.cc
@@ -87,9 +87,23 @@ std::string StrCat(const AlphaNum& a, const AlphaNum& b, const AlphaNum& c) {
       result, static_cast<size_t>(result_size),
       [&a, &b, &c](char* const begin, size_t buf_size) {
         char* out = begin;
-        out = Append(out, a);
-        out = Append(out, b);
-        out = Append(out, c);
+        char* out1 = nullptr;
+        char* out2 = nullptr;
+        char* out3 = nullptr;
+        out1 = out = Append(out, a);
+        out2 = out = Append(out, b);
+        out3 = out = Append(out, c);
+        if (out != begin + buf_size) {
+          fprintf(stderr, "begin=%p\n out1=%p\n out2=%p\n out3=%p\n a.size()=%lu\n b.size()=%lu\n, c.size()=%lu\n",
+              static_cast<void*>(begin),
+              static_cast<void*>(out1),
+              static_cast<void*>(out2),
+              static_cast<void*>(out3),
+              static_cast<size_t>(a.size()),
+              static_cast<size_t>(b.size()),
+              static_cast<size_t>(c.size())
+              );
+        }
         assert(out == begin + buf_size);
         return buf_size;
       });
