diff --git a/upb/reflection/BUILD b/upb/reflection/BUILD
index a980f8c6bd..489559a2fc 100644
--- a/upb/reflection/BUILD
+++ b/upb/reflection/BUILD
@@ -60,6 +60,12 @@ bootstrap_cc_library(
     ],
 )
 
+alias(
+    name = "reflection_cc",
+    actual = ":reflection",
+    visibility = ["//visibility:public"],
+)
+
 bootstrap_cc_library(
     name = "internal",
     srcs = [
@@ -138,6 +144,12 @@ bootstrap_cc_library(
     ],
 )
 
+alias(
+    name = "internal_cc",
+    actual = ":internal",
+    visibility = ["//upb:__subpackages__"],
+)
+
 cc_library(
     name = "generated_reflection_support",
     hdrs = [
