diff --git a/go/private/rules/binary.bzl b/go/private/rules/binary.bzl
index 40a17f4d..81f3ed86 100644
--- a/go/private/rules/binary.bzl
+++ b/go/private/rules/binary.bzl
@@ -462,8 +462,8 @@ exit /b %GO_EXIT_CODE%
             content = cmd,
         )
         ctx.actions.run(
-            executable = bat,
-            inputs = sdk.headers + sdk.tools + sdk.srcs + ctx.files.srcs + [sdk.go],
+            executable = bat.path.replace("/", "\\"),
+            inputs = sdk.libs + sdk.headers + sdk.tools + ctx.files.srcs + [sdk.go],
             outputs = [out, gotmp],
             mnemonic = "GoToolchainBinaryBuild",
         )
