diff --git a/MODULE.bazel b/MODULE.bazel
index 0110907f..d639012d 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -18,6 +18,7 @@ bazel_dep(name = "opentracing-cpp", version = "1.6.0", repo_name = "com_github_o
 bazel_dep(name = "platforms", version = "0.0.8")
 bazel_dep(name = "prometheus-cpp", version = "1.3.0", repo_name = "com_github_jupp0r_prometheus_cpp")
 bazel_dep(name = "protobuf", version = "26.0", repo_name = "com_google_protobuf")
+bazel_dep(name = "rules_cc", version = "0.2.19")
 bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
 bazel_dep(name = "zlib", version = "1.3.1.bcr.1")
 
diff --git a/api/BUILD b/api/BUILD
index dfc7d3c6..0ffdbc33 100644
--- a/api/BUILD
+++ b/api/BUILD
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 load("@bazel_skylib//rules:common_settings.bzl", "bool_flag", "int_flag", "string_flag")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 
 package(default_visibility = ["//visibility:public"])
 
diff --git a/api/test/baggage/BUILD b/api/test/baggage/BUILD
index a3783a63..c25f21e3 100644
--- a/api/test/baggage/BUILD
+++ b/api/test/baggage/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/baggage/propagation/BUILD b/api/test/baggage/propagation/BUILD
index 2b820ed3..889deacf 100644
--- a/api/test/baggage/propagation/BUILD
+++ b/api/test/baggage/propagation/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/common/BUILD b/api/test/common/BUILD
index 5ea2d5d5..3af58b53 100644
--- a/api/test/common/BUILD
+++ b/api/test/common/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 otel_cc_benchmark(
diff --git a/api/test/context/BUILD b/api/test/context/BUILD
index 64b2cb75..f7a10642 100644
--- a/api/test/context/BUILD
+++ b/api/test/context/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/context/propagation/BUILD b/api/test/context/propagation/BUILD
index d62d6764..68c01ae7 100644
--- a/api/test/context/propagation/BUILD
+++ b/api/test/context/propagation/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/core/BUILD b/api/test/core/BUILD
index 38c65e6d..ed0895a5 100644
--- a/api/test/core/BUILD
+++ b/api/test/core/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "timestamp_test",
     srcs = [
diff --git a/api/test/logs/BUILD b/api/test/logs/BUILD
index 521620a2..31ec9fd6 100644
--- a/api/test/logs/BUILD
+++ b/api/test/logs/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/metrics/BUILD b/api/test/metrics/BUILD
index 86e20033..9674ff90 100644
--- a/api/test/metrics/BUILD
+++ b/api/test/metrics/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/nostd/BUILD b/api/test/nostd/BUILD
index 4f2b1d31..e8485194 100644
--- a/api/test/nostd/BUILD
+++ b/api/test/nostd/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "function_ref_test",
     srcs = [
diff --git a/api/test/plugin/BUILD b/api/test/plugin/BUILD
index f20143ff..dc57d6f6 100644
--- a/api/test/plugin/BUILD
+++ b/api/test/plugin/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "dynamic_load_test",
     srcs = [
diff --git a/api/test/singleton/BUILD b/api/test/singleton/BUILD
index eae9f6b8..af24c17c 100644
--- a/api/test/singleton/BUILD
+++ b/api/test/singleton/BUILD
@@ -1,6 +1,10 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 # gcc and clang, assumed to be used on this platform
 DEFAULT_NOWIN_COPTS = [
     "-fvisibility=default",
diff --git a/api/test/trace/BUILD b/api/test/trace/BUILD
index d2330845..fd39ec9e 100644
--- a/api/test/trace/BUILD
+++ b/api/test/trace/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/trace/propagation/BUILD b/api/test/trace/propagation/BUILD
index 8e9718e8..285c2919 100644
--- a/api/test/trace/propagation/BUILD
+++ b/api/test/trace/propagation/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/api/test/trace/propagation/detail/BUILD b/api/test/trace/propagation/detail/BUILD
index 322c8929..7e73aadf 100644
--- a/api/test/trace/propagation/detail/BUILD
+++ b/api/test/trace/propagation/detail/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/bazel/extra_deps.bzl b/bazel/extra_deps.bzl
index 00a32c50..f68bc0f1 100644
--- a/bazel/extra_deps.bzl
+++ b/bazel/extra_deps.bzl
@@ -3,8 +3,8 @@
 
 # Load prometheus C++ dependencies.
 
-load("@com_github_jupp0r_prometheus_cpp//bazel:repositories.bzl", "prometheus_cpp_repositories")
 load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
+load("@com_github_jupp0r_prometheus_cpp//bazel:repositories.bzl", "prometheus_cpp_repositories")
 load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
 
 def opentelemetry_extra_deps():
diff --git a/bazel/otel_cc_benchmark.bzl b/bazel/otel_cc_benchmark.bzl
index cb181f83..5e4cd1b7 100644
--- a/bazel/otel_cc_benchmark.bzl
+++ b/bazel/otel_cc_benchmark.bzl
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 def otel_cc_benchmark(name, srcs, deps, tags = [""]):
     """
     Creates targets for the benchmark and related targets.
@@ -22,7 +25,7 @@ def otel_cc_benchmark(name, srcs, deps, tags = [""]):
 
     # This is the benchmark as a binary, it can be run manually, and is used
     # to generate the _result below.
-    native.cc_binary(
+    cc_binary(
         name = name,
         srcs = srcs,
         deps = deps + ["@com_github_google_benchmark//:benchmark"],
@@ -42,7 +45,7 @@ def otel_cc_benchmark(name, srcs, deps, tags = [""]):
 
     # This is run as part of "bazel test ..." to smoke-test benchmarks. It's
     # meant to complete quickly rather than get accurate results.
-    native.cc_test(
+    cc_test(
         name = name + "_smoketest",
         srcs = srcs,
         deps = deps + ["@com_github_google_benchmark//:benchmark"],
diff --git a/examples/batch/BUILD b/examples/batch/BUILD
index c5db0f4b..557b5b6b 100644
--- a/examples/batch/BUILD
+++ b/examples/batch/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_simple",
     srcs = [
diff --git a/examples/common/foo_library/BUILD b/examples/common/foo_library/BUILD
index 541d7869..684032cd 100644
--- a/examples/common/foo_library/BUILD
+++ b/examples/common/foo_library/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/examples/common/logs_foo_library/BUILD b/examples/common/logs_foo_library/BUILD
index 3dc0ae9f..644ecd15 100644
--- a/examples/common/logs_foo_library/BUILD
+++ b/examples/common/logs_foo_library/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/examples/common/metrics_foo_library/BUILD b/examples/common/metrics_foo_library/BUILD
index 55ab0154..90928d1a 100644
--- a/examples/common/metrics_foo_library/BUILD
+++ b/examples/common/metrics_foo_library/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/examples/etw_threads/BUILD b/examples/etw_threads/BUILD
index edb60080..1722c1a5 100644
--- a/examples/etw_threads/BUILD
+++ b/examples/etw_threads/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_etw_threads",
     srcs = [
diff --git a/examples/grpc/BUILD b/examples/grpc/BUILD
index b1d04e9f..94e19f7e 100644
--- a/examples/grpc/BUILD
+++ b/examples/grpc/BUILD
@@ -1,10 +1,13 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
-package(default_visibility = ["//visibility:public"])
-
-load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+package(default_visibility = ["//visibility:public"])
 
 proto_library(
     name = "messages_proto",
diff --git a/examples/http/BUILD b/examples/http/BUILD
index 6e98ac03..3a4b0114 100644
--- a/examples/http/BUILD
+++ b/examples/http/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_http_client",
     srcs = [
diff --git a/examples/logs_simple/BUILD b/examples/logs_simple/BUILD
index 03073b42..0481b1cf 100644
--- a/examples/logs_simple/BUILD
+++ b/examples/logs_simple/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_logs_simple",
     srcs = [
diff --git a/examples/metrics_simple/BUILD b/examples/metrics_simple/BUILD
index 029fda8a..b1042ad4 100644
--- a/examples/metrics_simple/BUILD
+++ b/examples/metrics_simple/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "metrics_ostream_example",
     srcs = [
diff --git a/examples/multi_processor/BUILD b/examples/multi_processor/BUILD
index 1c7a3789..c9125891 100644
--- a/examples/multi_processor/BUILD
+++ b/examples/multi_processor/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_multi_processor",
     srcs = [
diff --git a/examples/multithreaded/BUILD b/examples/multithreaded/BUILD
index 4ee13862..4c40c163 100644
--- a/examples/multithreaded/BUILD
+++ b/examples/multithreaded/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_multithreaded",
     srcs = [
diff --git a/examples/otlp/BUILD b/examples/otlp/BUILD
index b27cbef3..b346630b 100644
--- a/examples/otlp/BUILD
+++ b/examples/otlp/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_otlp_grpc",
     srcs = [
diff --git a/examples/plugin/load/BUILD b/examples/plugin/load/BUILD
index b129854a..82e571ad 100644
--- a/examples/plugin/load/BUILD
+++ b/examples/plugin/load/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "load_plugin",
     srcs = [
diff --git a/examples/plugin/plugin/BUILD b/examples/plugin/plugin/BUILD
index 2f739806..3d1f129b 100644
--- a/examples/plugin/plugin/BUILD
+++ b/examples/plugin/plugin/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_plugin.so",
     srcs = [
diff --git a/examples/prometheus/BUILD b/examples/prometheus/BUILD
index 0a6789f2..45489886 100644
--- a/examples/prometheus/BUILD
+++ b/examples/prometheus/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "prometheus_example",
     srcs = [
diff --git a/examples/simple/BUILD b/examples/simple/BUILD
index 5a3c27c2..c155a736 100644
--- a/examples/simple/BUILD
+++ b/examples/simple/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "example_simple",
     srcs = [
diff --git a/exporters/elasticsearch/BUILD b/exporters/elasticsearch/BUILD
index 9990c258..71ff5f39 100644
--- a/exporters/elasticsearch/BUILD
+++ b/exporters/elasticsearch/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/exporters/etw/BUILD b/exporters/etw/BUILD
index ad8518bc..fa121d99 100644
--- a/exporters/etw/BUILD
+++ b/exporters/etw/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/exporters/memory/BUILD b/exporters/memory/BUILD
index 033c85d3..e7a94ee7 100644
--- a/exporters/memory/BUILD
+++ b/exporters/memory/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/exporters/ostream/BUILD b/exporters/ostream/BUILD
index 177bf6a9..0039529b 100644
--- a/exporters/ostream/BUILD
+++ b/exporters/ostream/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/exporters/otlp/BUILD b/exporters/otlp/BUILD
index cdaba37e..5771e4e3 100644
--- a/exporters/otlp/BUILD
+++ b/exporters/otlp/BUILD
@@ -1,10 +1,12 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
-package(default_visibility = ["//visibility:public"])
-
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
+package(default_visibility = ["//visibility:public"])
+
 cc_library(
     name = "otlp_recordable",
     srcs = [
diff --git a/exporters/prometheus/BUILD b/exporters/prometheus/BUILD
index fc981921..c6c0cf9e 100644
--- a/exporters/prometheus/BUILD
+++ b/exporters/prometheus/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/exporters/zipkin/BUILD b/exporters/zipkin/BUILD
index 16774e80..bfa91014 100644
--- a/exporters/zipkin/BUILD
+++ b/exporters/zipkin/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/ext/BUILD b/ext/BUILD
index b19ae921..52ed22af 100644
--- a/ext/BUILD
+++ b/ext/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/ext/src/http/client/curl/BUILD b/ext/src/http/client/curl/BUILD
index de5a5aeb..0dc5d7e7 100644
--- a/ext/src/http/client/curl/BUILD
+++ b/ext/src/http/client/curl/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/ext/test/http/BUILD b/ext/test/http/BUILD
index d030ec28..c7d1dfde 100644
--- a/ext/test/http/BUILD
+++ b/ext/test/http/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "curl_http_test",
     srcs = [
diff --git a/ext/test/w3c_tracecontext_http_test_server/BUILD b/ext/test/w3c_tracecontext_http_test_server/BUILD
index 8c1c8c17..5c3b1971 100644
--- a/ext/test/w3c_tracecontext_http_test_server/BUILD
+++ b/ext/test/w3c_tracecontext_http_test_server/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "w3c_tracecontext_http_test_server",
     srcs = [
diff --git a/opentracing-shim/BUILD b/opentracing-shim/BUILD
index e7d90a4b..6daa05b6 100644
--- a/opentracing-shim/BUILD
+++ b/opentracing-shim/BUILD
@@ -1,6 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/BUILD b/sdk/BUILD
index b19ae921..52ed22af 100644
--- a/sdk/BUILD
+++ b/sdk/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/common/BUILD b/sdk/src/common/BUILD
index 9c424cac..b02b837b 100644
--- a/sdk/src/common/BUILD
+++ b/sdk/src/common/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/common/platform/BUILD b/sdk/src/common/platform/BUILD
index f089ebf7..bf4deb17 100644
--- a/sdk/src/common/platform/BUILD
+++ b/sdk/src/common/platform/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/logs/BUILD b/sdk/src/logs/BUILD
index 93642e99..bfca6ff8 100644
--- a/sdk/src/logs/BUILD
+++ b/sdk/src/logs/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/metrics/BUILD b/sdk/src/metrics/BUILD
index 78f224b5..f1fb4efd 100644
--- a/sdk/src/metrics/BUILD
+++ b/sdk/src/metrics/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/resource/BUILD b/sdk/src/resource/BUILD
index 88456299..e73e4c05 100644
--- a/sdk/src/resource/BUILD
+++ b/sdk/src/resource/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/src/trace/BUILD b/sdk/src/trace/BUILD
index df99eb15..a41637b0 100644
--- a/sdk/src/trace/BUILD
+++ b/sdk/src/trace/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/sdk/test/common/BUILD b/sdk/test/common/BUILD
index de4c02ec..2d792d4a 100644
--- a/sdk/test/common/BUILD
+++ b/sdk/test/common/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/sdk/test/instrumentationscope/BUILD b/sdk/test/instrumentationscope/BUILD
index 39a61564..34b5e0f4 100644
--- a/sdk/test/instrumentationscope/BUILD
+++ b/sdk/test/instrumentationscope/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "instrumentationscope_test",
     srcs = [
diff --git a/sdk/test/logs/BUILD b/sdk/test/logs/BUILD
index 2138a448..6baa7b3e 100644
--- a/sdk/test/logs/BUILD
+++ b/sdk/test/logs/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "logger_provider_sdk_test",
     srcs = [
diff --git a/sdk/test/metrics/BUILD b/sdk/test/metrics/BUILD
index 70d0cc06..00861749 100644
--- a/sdk/test/metrics/BUILD
+++ b/sdk/test/metrics/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_library(
diff --git a/sdk/test/metrics/exemplar/BUILD b/sdk/test/metrics/exemplar/BUILD
index 413780b7..9b11a64b 100644
--- a/sdk/test/metrics/exemplar/BUILD
+++ b/sdk/test/metrics/exemplar/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "no_exemplar_reservoir_test",
     srcs = [
diff --git a/sdk/test/resource/BUILD b/sdk/test/resource/BUILD
index 145ecf0b..92005f3a 100644
--- a/sdk/test/resource/BUILD
+++ b/sdk/test/resource/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "resource_test",
     srcs = [
diff --git a/sdk/test/trace/BUILD b/sdk/test/trace/BUILD
index 9aa8a95c..832cc74e 100644
--- a/sdk/test/trace/BUILD
+++ b/sdk/test/trace/BUILD
@@ -1,6 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:otel_cc_benchmark.bzl", "otel_cc_benchmark")
 
 cc_test(
diff --git a/test_common/BUILD b/test_common/BUILD
index b19ae921..52ed22af 100644
--- a/test_common/BUILD
+++ b/test_common/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/test_common/src/http/client/nosend/BUILD b/test_common/src/http/client/nosend/BUILD
index fa7ba623..9436ec00 100644
--- a/test_common/src/http/client/nosend/BUILD
+++ b/test_common/src/http/client/nosend/BUILD
@@ -1,6 +1,8 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git a/tools/WORKSPACE b/tools/WORKSPACE
index 3e90ba0c..404f3179 100644
--- a/tools/WORKSPACE
+++ b/tools/WORKSPACE
@@ -2,6 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 local_repository(
-  name = "vcpkg",
-  path = "./vcpkg",
+    name = "vcpkg",
+    path = "./vcpkg",
 )
