diff --git a/upb_generator/c/names.cc b/upb_generator/c/names.cc
index ddca7d7fa6210..5dcf1eeec3d3e 100644
--- a/upb_generator/c/names.cc
+++ b/upb_generator/c/names.cc
@@ -69,7 +69,7 @@ constexpr uint32_t kAnyField = UINT32_MAX;
 
 // Prefixes used by C code generator for field access.
 static constexpr std::array<Prefix, 6> kPrefixes{
-    Prefix{"clear_", kContainerField | kStringField},
+    Prefix{"clear_", kAnyField},
     Prefix{"delete_", kContainerField},
     Prefix{"add_", kContainerField},
     Prefix{"resize_", kContainerField},
