diff --git a/upb/port/def.inc b/upb/port/def.inc
index 4616c65cc..ba5d01e6d 100644
--- a/upb/port/def.inc
+++ b/upb/port/def.inc
@@ -668,7 +668,7 @@ Error, UINTPTR_MAX is undefined
  * to prevent optimizers from removing the constructor. Our solution is to
  * create a dummy exported weak symbol that prevent this stripping.
  */
-#pragma section(".CRT$XCU", long, read)
+#pragma section(".CRT$XCU", write, read)
 #define UPB_CONSTRUCTOR(name, unique_name)                                   \
   static void __cdecl UPB_PRIVATE(name)(void);                               \
   __declspec(allocate(".CRT$XCU"), selectany) void(                          \
