diff --git a/upb/port/def.inc b/upb/port/def.inc
index 49e469990..2c4b2029d 100644
--- a/upb/port/def.inc
+++ b/upb/port/def.inc
@@ -578,35 +578,6 @@ Error, UINTPTR_MAX is undefined
 #define UPB_LINKARR_START(name) (&__start_linkarr_##name)
 #define UPB_LINKARR_STOP(name) (&__stop_linkarr_##name)
 
-#elif defined(_MSC_VER)
-
-/* See:
- *   https://devblogs.microsoft.com/oldnewthing/20181107-00/?p=100155
- *   https://devblogs.microsoft.com/oldnewthing/20181108-00/?p=100165
- *   https://devblogs.microsoft.com/oldnewthing/20181109-00/?p=100175 */
-#define UPB_STRINGIFY_INTERNAL(x) #x
-#define UPB_STRINGIFY(x) UPB_STRINGIFY_INTERNAL(x)
-#define UPB_CONCAT(a, b, c) a##b##c
-#define UPB_LINKARR_NAME(name, index) \
-  UPB_STRINGIFY(UPB_CONCAT(la_, name, index))
-#define UPB_LINKARR_APPEND(name)                      \
-  __pragma(section(UPB_LINKARR_NAME(name, $j), read)) \
-      __declspec(allocate(UPB_LINKARR_NAME(name, $j)))
-// clang-format off
-#define UPB_LINKARR_DECLARE(name, type)                          \
-  __pragma(message(UPB_LINKARR_NAME(name, $j)))                  \
-  __pragma(section(UPB_LINKARR_NAME(name, $a), read))            \
-  __pragma(section(UPB_LINKARR_NAME(name, $z), read))            \
-  __declspec(allocate(UPB_LINKARR_NAME(name, $a)), selectany)    \
-            type __start_linkarr_##name;                         \
-  __declspec(allocate(UPB_LINKARR_NAME(name, $z)), selectany)    \
-            type __stop_linkarr_##name;                          \
-  UPB_LINKARR_APPEND(name)                                       \
-  __declspec(selectany) type UPB_linkarr_internal_empty_##name[1] = {0}
-// clang-format on
-#define UPB_LINKARR_START(name) (&__start_linkarr_##name)
-#define UPB_LINKARR_STOP(name) (&__stop_linkarr_##name)
-
 #else
 
 // Linker arrays are not supported on this platform.  Make macros no-ops.
@@ -633,24 +604,6 @@ Error, UINTPTR_MAX is undefined
   _UPB_CONSTRUCTOR_PLACEHOLDER(unique_name)                                  \
   __attribute__((weak, visibility("hidden"), constructor)) void UPB_PRIVATE( \
       name)(void)
-#elif defined(_MSC_VER)
-/*
- * See: https://stackoverflow.com/questions/1113409
- *
- * The /include pragma suggested in the link above doesn't work in our case
- * because it requires globally unique names. We need a different solution
- * to prevent optimizers from removing the constructor. Our solution is to
- * create a dummy exported weak symbol that prevent this stripping.
- */
-#pragma section(".CRT$XCU", long, read)
-#define UPB_CONSTRUCTOR(name, unique_name)                                   \
-  static void __cdecl UPB_PRIVATE(name)(void);                               \
-  __declspec(allocate(".CRT$XCU"), selectany) void(                          \
-      __cdecl * UPB_PRIVATE(name##_))(void) = UPB_PRIVATE(name);             \
-  __declspec(selectany, dllexport) void* UPB_PRIVATE(name##_force_linkage) = \
-      &UPB_PRIVATE(name##_);                                                 \
-  static void __cdecl UPB_PRIVATE(name)(void)
-
 #else
 // No constructor support, nothing we can do except not break builds.
 #define UPB_CONSTRUCTOR(name, unique_name) static void UPB_PRIVATE(name)(void)
