diff --git a/build_defs/BUILD.bazel b/build_defs/BUILD.bazel
index 1c72eba68..176ab5a91 100644
--- a/build_defs/BUILD.bazel
+++ b/build_defs/BUILD.bazel
@@ -14,10 +14,24 @@ package(
 )
 
 create_compiler_config_setting(
-    name = "config_msvc",
+    name = "config_msvc_cl",
     value = "msvc-cl",
 )
 
+# Caveat: clang-cl support in protobuf is only best-effort / untested for now.
+create_compiler_config_setting(
+    name = "config_clang_cl",
+    value = "clang-cl",
+)
+
+selects.config_setting_group(
+    name = "config_msvc",
+    match_any = [
+        ":config_clang_cl",
+        ":config_msvc_cl",
+    ],
+)
+
 config_setting(
     name = "aarch64",
     values = {"cpu": "linux-aarch_64"},
