diff --git a/protobuf_deps.bzl b/protobuf_deps.bzl
index b8754c379..b6af7f09b 100644
--- a/protobuf_deps.bzl
+++ b/protobuf_deps.bzl
@@ -72,11 +72,14 @@ def protobuf_deps():
         )
 
     if not native.existing_rule("abseil-cpp"):
-        _github_archive(
+        http_archive(
             name = "abseil-cpp",
-            repo = "https://github.com/abseil/abseil-cpp",
-            commit = "9ac7062b1860d895fb5a8cbf58c3e9ef8f674b5f",  # Abseil LTS 20250127
-            sha256 = "d8ae9aa794a571ee39c77085ee69f1d4ac276212a7d99734974d95df7baa8d13",
+            sha256 = "9b7a064305e9fd94d124ffa6cc358592eb42b5da588fb4e07d09254aa40086db",
+            strip_prefix = "abseil-cpp-20250512.1",
+            urls = [
+                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/abseil/abseil-cpp/archive/refs/tags/20250512.1.tar.gz",
+                "https://github.com/abseil/abseil-cpp/archive/refs/tags/20250512.1.tar.gz",
+            ],
         )
 
     if not native.existing_rule("zlib"):
diff --git a/python/google/protobuf/__init__.py b/python/google/protobuf/__init__.py
index d41e5bf38..741d75d8d 100755
--- a/python/google/protobuf/__init__.py
+++ b/python/google/protobuf/__init__.py
@@ -8,3 +8,9 @@
 # Copyright 2007 Google Inc. All Rights Reserved.
 
 __version__ = '6.31.1'
+
+if __name__ != '__main__':
+  try:
+    __import__('pkg_resources').declare_namespace(__name__)
+  except ImportError:
+    __path__ = __import__('pkgutil').extend_path(__path__, __name__)
