diff --git a/bazel/py_proto_library.bzl b/bazel/py_proto_library.bzl
index 86af99950..8396661ae 100644
--- a/bazel/py_proto_library.bzl
+++ b/bazel/py_proto_library.bzl
@@ -147,6 +147,9 @@ def _py_proto_library_rule(ctx):
         transitive = [info.transitive_sources for info in pyproto_infos],
     )
 
+    # Gather imports from the 'imports' attribute
+    imports = depset(direct = [ctx.label.package + "/" + i for i in ctx.attr.imports])
+
     return [
         DefaultInfo(
             files = default_outputs,
@@ -161,7 +164,8 @@ def _py_proto_library_rule(ctx):
         ),
         PyInfo(
             transitive_sources = default_outputs,
-            imports = depset(transitive = [info.imports for info in pyproto_infos]),
+            # Include imports from the 'imports' attribute and from dependencies
+            imports = depset(transitive = [info.imports for info in pyproto_infos] + [imports]),
             # Proto always produces 2- and 3- compatible source files
             has_py2_only_sources = False,
             has_py3_only_sources = False,
@@ -206,6 +210,7 @@ proto_library(
             providers = [ProtoInfo],
             aspects = [_py_proto_aspect],
         ),
+        "imports": attr.string_list(),
     },
     provides = [PyInfo],
 )
